File: Front-End/src/app/models/workshop/tool.control/tool.control.category.ts
import { StatusEnabDisabEnum } from "../../status-enab-disab-enum";
import { TypeCategoryEnum } from "../type.category.enum";

export class ToolControlCategory {
    companyId: number | null = null;
    resaleId: number | null = null;
    id: number | null = null;
    status: StatusEnabDisabEnum = StatusEnabDisabEnum.DISABLED;
    type: TypeCategoryEnum = TypeCategoryEnum.OTHER;
    quantityReq: number = 0;
    description: string = '';
}

File: API/src/main/java/com/concierge/apiconcierge/models/workshop/toolcontrol/ToolControlCategory.java
package com.concierge.apiconcierge.models.workshop.toolcontrol;

import com.concierge.apiconcierge.models.enums.StatusEnableDisable;
import com.concierge.apiconcierge.models.workshop.toolcontrol.enums.TypeCategory;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
@Entity
@Table(name = "tb_tool_control_category")
public class ToolControlCategory {

    @Column(name = "company_id")
    private Integer companyId;

    @Column(name = "resale_id")
    private Integer resaleId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private StatusEnableDisable status;

    private TypeCategory type;

    @Column(name = "quantity_req")
    private Integer quantityReq;

    private String description;
}
